### ISSUE TITLE
Can't recreate MSE as custom loss

### ISSUE THREAD

**Issue by han-ol** [google/yggdrasil-decision-forests#189]
Nice work with this library, thank you!

Following the tutorial on custom losses, I can't recreate the MSE loss.

What seems to be the issue here?

# Minimal example

```python
import ydf
import numpy as np
import pandas as pd  # We use Pandas to load small datasets

# Download a classification dataset and load it as a Pandas DataFrame.
ds_path = "https://raw.githubusercontent.com/google/yggdrasil-decision-forests/main/yggdrasil_decision_forests/test_data/dataset"
all_ds = pd.read_csv(f"{ds_path}/abalone.csv")

# Randomly split the dataset into a training (70%) and testing (30%) dataset
all_ds = all_ds.sample(frac=1)
split_idx = len(all_ds) * 7 // 10
train_ds = all_ds.iloc[:split_idx]
test_ds = all_ds.iloc[split_idx:]

def loss_mse(labels, predictions, weights=None):  # careful, weights are ignored, but not the focus here
    return np.mean((predictions - labels)**2)

def initial_predictions_mse(labels, weights):
    return np.mean(labels)

def grad_mse(labels, predictions):
    return 2.0 / len(labels) * (predictions - labels)

def hessian_mse(labels, predictions):
    return np.full_like(predictions, 2.0 / len(labels))

def gradient_and_hessian_mse(labels, predictions):
    return grad_mse(labels, predictions), hessian_mse(labels, predictions)

mse_custom_loss = ydf.RegressionLoss(
    initial_predictions=initial_predictions_mse,
    gradient_and_hessian=gradient_and_hessian_mse,
    loss=loss_mse,
    activation=ydf.Activation.IDENTITY,
)

model = ydf.GradientBoostedTreesLearner(label="Rings", loss=mse_custom_loss,
                                task=ydf.Task.REGRESSION).train(train_ds)

evaluation
```
The model predictions blow up, see for example 
RMSE:
8.3044e+13

# Expected behavior

I expect to reproduce the behavior of not specifying the loss argument perfectly, as the default loss should also be MSE and my custom_mse_loss seems specified correctly (?).

I like the ability to add custom losses and would really appreciate if you could look into what goes sideways here.

# Evaluation figure

![Image](https://github.com/user-attachments/assets/cf9dae16-b5e6-4fcf-af29-031fc66c4461)

**Comment by rstz** [comment]
You need the hessian to be negative. This is a quirk in YDF that we should probably document better.

For reference, this is a test that reproduces exactly the internal MSE loss with a custom loss https://github.com/google/yggdrasil-decision-forests/blob/main/yggdrasil_decision_forests/port/python/ydf/learner/custom_loss_test.py#L203

**Comment by han-ol** [comment]
Yes, flipping the hessian sign made it work as expected. Thanks!

**Comment by rstz** [comment]
Great!
